### PR TITLE
[Track2-Kudu-2] Extract abstract app service in app service library

### PR DESCRIPTION
**Pull request by Flanker32**
Refactor library and extract abstract app service for common logic

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/service/impl/WebApp.java [hunk 1]
```diff
@@ -28,19 +25,18 @@
...
 import java.util.stream.Collectors;
 
-public class WebApp implements IWebApp {
+public class WebApp extends AbstractAppService<com.azure.resourcemanager.appservice.models.WebApp> implements IWebApp {
 
     private static final String UNSUPPORTED_OPERATING_SYSTEM = "Unsupported operating system %s";
     private WebAppEntity entity;
 
-    private AzureResourceManager azureClient;
+    private final AzureResourceManager azureClient;
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/service/impl/WebApp.java:39]: its confusing to have: impl.WebApp, model.WebApp and WebAppEntity

Comment (Flanker32) [reply azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/service/impl/WebApp.java:39]: `model.WebApp` is the interface of track2 sdk, rename it to remote

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/service/impl/AbstractAppService.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import com.microsoft.azure.toolkit.lib.appservice.model.Runtime;
+import com.microsoft.azure.toolkit.lib.appservice.service.IAppService;
+import com.microsoft.azure.toolkit.lib.common.exception.AzureToolkitRuntimeException;
+import reactor.core.publisher.Flux;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Optional;
+
+public abstract class AbstractAppService<T extends WebAppBase> implements IAppService {
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/service/impl/AbstractAppService.java:21]: should not be public

**Review by andxu — APPROVED** [review]

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/service/impl/WebAppDeploymentSlot.java [hunk 3]
```diff
@@ -59,105 +56,36 @@ public IWebAppDeploymentSlotUpdater update() {
...
-    @Override
-    public String id() {
-        return getDeploymentSlotInner().id();
-    }
-
-    @Override
-    public String name() {
-        return getDeploymentSlotInner().name();
+        return StringUtils.isNotEmpty(entity.getId()) ?
+                azureClient.webApps().getById(entity.getId().substring(0, entity.getId().indexOf("/slots"))) :
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/service/impl/WebAppDeploymentSlot.java:87]: dangerous if  rg starts with slots

Comment (Flanker32) [reply azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/service/impl/WebAppDeploymentSlot.java:87]: Will fix it in later pr

**Pull request merged**